src/backend_common.cc

```diff
@@ -201,6 +201,53 @@ GetByteSize(
   return cnt * dt_size;
 }
 
+TRITONSERVER_Error*
+ReadInputTensor(
+    TRITONBACKEND_Request* request, const std::string& input_name, char* buffer,
+    size_t* buffer_byte_size, TRITONSERVER_MemoryType memory_type,
+    int64_t memory_type_id, cudaStream_t cuda_stream, bool* cuda_used,
+    const char* host_policy_name, const bool copy_on_stream)
+{
+  TRITONBACKEND_Input* input;
```

Comment:
A lot of code duplication can be reduced by using this new function signature within the one with original signature.